dealii@develop  build failure with boost@1.70.0 on cori

I'm seeing a build failure of  dealii@develop  with boost@1.70.0 on cori

The same build succeeds with boost@1.69.0.

Attaching spack-build.out for both builds

[dealii@develop-with-boost@1.69.0-spack-build.out.txt](https://github.com/dealii/dealii/files/3094568/dealii%40develop-with-boost%401.69.0-spack-build.out.txt)
[dealii@develop-with-boost@1.70.0-spack-build.out.txt](https://github.com/dealii/dealii/files/3094569/dealii%40develop-with-boost%401.70.0-spack-build.out.txt)

The relevant error message in the second log file is this:
```
-- Include /global/u2/b/balay/spack/var/spack/stage/dealii-develop-gui2wav57bgxsr2zehgd2bmgncamf2e5/dealii/cmake/configure/configure_2_boost.cmake
-- Found Boost 1.70.0 at /global/u2/b/balay/spack/opt/spack/cray-cnl9-haswell/intel-19.0.3.199/boost-1.70.0-ypmbbtkavrpxewjj6yqi6etotlsbw2on/lib/cmake/Boost-1.70.0
--   Requested configuration: QUIET COMPONENTS iostreams;serialization;system;thread
-- Found boost_headers 1.70.0 at /global/u2/b/balay/spack/opt/spack/cray-cnl9-haswell/intel-19.0.3.199/boost-1.70.0-ypmbbtkavrpxewjj6yqi6etotlsbw2on/lib/cmake/boost_headers-1.70.0
-- Found boost_iostreams 1.70.0 at /global/u2/b/balay/spack/opt/spack/cray-cnl9-haswell/intel-19.0.3.199/boost-1.70.0-ypmbbtkavrpxewjj6yqi6etotlsbw2on/lib/cmake/boost_iostreams-1.70.0
--   libboost_iostreams.so.1.70.0
-- Adding boost_iostreams dependencies: headers
-- Found boost_serialization 1.70.0 at /global/u2/b/balay/spack/opt/spack/cray-cnl9-haswell/intel-19.0.3.199/boost-1.70.0-ypmbbtkavrpxewjj6yqi6etotlsbw2on/lib/cmake/boost_serialization-1.70.0
--   libboost_serialization.so.1.70.0
-- Adding boost_serialization dependencies: headers
-- Found boost_system 1.70.0 at /global/u2/b/balay/spack/opt/spack/cray-cnl9-haswell/intel-19.0.3.199/boost-1.70.0-ypmbbtkavrpxewjj6yqi6etotlsbw2on/lib/cmake/boost_system-1.70.0
--   libboost_system.so.1.70.0
-- Adding boost_system dependencies: headers
-- Found boost_thread 1.70.0 at /global/u2/b/balay/spack/opt/spack/cray-cnl9-haswell/intel-19.0.3.199/boost-1.70.0-ypmbbtkavrpxewjj6yqi6etotlsbw2on/lib/cmake/boost_thread-1.70.0
--   libboost_thread.so.1.70.0
-- Adding boost_thread dependencies: headers
-- Boost 1.59 found.
-- Found Boost components:
   iostreams;serialization;system;thread
--   BOOST_VERSION: ..
--   BOOST_LIBRARIES: *** Required variable "Boost_LIBRARIES" empty ***
--   BOOST_INCLUDE_DIRS: *** Required variable "Boost_INCLUDE_DIRS" empty ***
--   BOOST_USER_INCLUDE_DIRS: *** Required variable "Boost_INCLUDE_DIRS" empty ***
-- Could NOT find BOOST
-- DEAL_II_WITH_BOOST has unmet external dependencies.
CMake Error at cmake/macros/macro_configure_feature.cmake:112 (MESSAGE):
  

  Could not find the boost library!

  Please ensure that a suitable boost library is installed on your computer.

  If the library is not at a default location, either provide some hints for
  autodetection,

      $ BOOST_DIR="..." cmake <...>
      $ cmake -DBOOST_DIR="..." <...>

  or set the relevant variables by hand in ccmake.

  Alternatively you may choose to compile the bundled library of boost by
  setting DEAL_II_ALLOW_BUNDLED=on or DEAL_II_FORCE_BUNDLED_BOOST=on.
```

--- a/cmake/modules/FindBOOST.cmake
+++ b/cmake/modules/FindBOOST.cmake
@@ -50,6 +50,10 @@ ENDIF()
 
 # temporarily disable ${CMAKE_SOURCE_DIR}/cmake/modules for module lookup
 LIST(REMOVE_ITEM CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
+
+# Workaround a CMake compatibility issue with boost-1.70.0
+SET(Boost_NO_BOOST_CMAKE ON)
+
 IF(DEAL_II_WITH_ZLIB)
   FIND_PACKAGE(Boost ${BOOST_VERSION_REQUIRED} COMPONENTS
     iostreams serialization system thread
@@ -77,7 +81,7 @@ IF(NOT Boost_FOUND AND Boost_USE_STATIC_LIBS)
   LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
 ENDIF()
 
-
+UNSET(Boost_NO_BOOST_CMAKE)
 
 IF(Boost_FOUND)
   #

--- a/cmake/modules/FindBOOST.cmake
+++ b/cmake/modules/FindBOOST.cmake
@@ -51,7 +51,9 @@ ENDIF()
 # temporarily disable ${CMAKE_SOURCE_DIR}/cmake/modules for module lookup
 LIST(REMOVE_ITEM CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
 
-# Workaround a CMake compatibility issue with boost-1.70.0
+# Work around a CMake compatibility issue with boost-1.70.0
+# compare https://gitlab.kitware.com/cmake/cmake/issues/18865
+# and https://lists.boost.org/Archives/boost/2019/02/245016.php
 SET(Boost_NO_BOOST_CMAKE ON)
 
 IF(DEAL_II_WITH_ZLIB)

